jsonargparse/_typehints.py

```diff
@@ -1101,14 +1101,15 @@ def adapt_typehints(
 
 
 def implements_protocol(value, protocol) -> bool:
+    allowed_dunder_methods = {"__call__"}
     from jsonargparse._parameter_resolvers import get_signature_parameters
     from jsonargparse._postponed_annotations import get_return_type
 
-    if not inspect.isclass(value):
+    if not inspect.isclass(value):  # Should we check if callables implement protocols?
```

Comment:
It does make sense that a function could implement a callable protocol. But I would leave this for a different pull request.

Reply:
Shall I open an issue so we can keep track of this feature/bug?

Reply:
Sure, open an issue.